frontend/src/pages/Products/components/ProductGrid.tsx

```diff
@@ -1,42 +1,20 @@
-import { Typography, Grid, Card, CardContent, CardMedia } from "@mui/material";
-import products from "../../../constants/products";
-import productStyles from "./styles";
+import Grid from "@mui/material/Grid2";
 import { Link } from "react-router-dom";
+import products from "../../../constants/products";
+import ProductCard from "./ProductCard";
 
 const ProductGrid: React.FC = () => {
     return (
         <Grid container spacing={4}>
             {products.map((product, index) => (
-                <Grid item key={index} xs={12} sm={6} md={4}>
+                <Grid key={index} size={{ xs: 2, sm: 4, md: 4 }}>
```

Comment:
Make sure that keys in lists are unique and meaningful. Currently, you are using index as the key, which is not ideal if the list can change.